content/en/community/hackathons/sessions/bincompat/content/Work-Items_02.-Build-app-elfloader.md

```diff
@@ -1,40 +1,76 @@
-Using `do.sh run` we ran the prebuilt images from [the `run-app-elfloader` repository](https://github.com/unikraft/run-app-elfloader).
+Using `./run*.sh` we ran the prebuilt images from [the `run-app-elfloader` repository](https://github.com/unikraft/run-app-elfloader).
 Let's now also build `app-elfloader`.
 
-Run the following commands:
+Run the following commands, similar to what we did in previous sessions when building usual applications:
 
 ```console
-$ ./do.sh clean
-
-$ ./do.sh setup
-
-$ ./do.sh build
-
-$ ./do.sh run_built ...
+$ git clone https://github.com/unikraft/app-elfloader/
+$ git clone https://github.com/unikraft/unikraft .unikraft/unikraft
+$ git clone https://github.com/unikraft/lib-lwip .unikraft/libs/lwip
+$ git clone https://github.com/unikraft/lib-libelf .unikraft/libs/libelf
+
+$ UK_DEFCONFIG=$(pwd)/.config.elfloader_qemu-x86_64 make defconfig
+/usr/bin/gcc -ldl -I. -I/tmp/usoc/app-elfloader/build/kconfig -DCONFIG_=\"\"   -c fixdep.c -o /tmp/usoc/app-elfloader/build/kconfig/fixdep.o
+/usr/bin/gcc -ldl -I. -I/tmp/usoc/app-elfloader/build/kconfig -DCONFIG_=\"\"   /tmp/usoc/app-elfloader/build/kconfig/fixdep.o -o /tmp/usoc/app-elfloader/build/kconfig/fixdep
+#
+# configuration written to /tmp/usoc/app-elfloader/.config
+#
+make[1]: Leaving directory '/tmp/usoc/app-elfloader/.unikraft/unikraft'
+
+$ make -j $(nproc)
+[...]
+  UKBI    elfloader_qemu-x86_64.dbg.bootinfo
+  SCSTRIP elfloader_qemu-x86_64
+  GZ      elfloader_qemu-x86_64.gz
+make[1]: Leaving directory '/tmp/usoc/app-elfloader/.unikraft/unikraft'
 ```
 
-In the last command, replace `...` with the name of the application you need to run.
-
-If you want to remove all the pesky debugging information, use `setup_plain`, such as the commands below:
+This will result, as we already know, in a `app-elfloader_qemu-x86_64` kernel image.
+To run an application using this `elfloader` image, we can still use the scripts from `run-app-elfloader`.
+The `run.sh` script can use a custom `elfloader` image with the `-k` option.
+Note that at the beginning of the sessions, we cloned the `app-elfloader`, `run-app-elfloader`, `dynamic-apps` and `static-pie-apps` repositories in the same root directory:
 
 ```console
-$ ./do.sh clean
-
-$ ./do.sh setup_plain
-
-$ ./do.sh build
-
-$ ./do.sh run_built ...
+$ tree -L 1 .
+.
+|-- app-elfloader/
+|-- dynamic-apps/
+`-- run-app-elfloader/
 ```
 
-If, on the other side, you want to have more debugging information, use `setup_debug`, such as the commands below:
-
 ```console
-$ ./do.sh clean
-
-$ ./do.sh setup_debug
+$ cd run-app-elfloader/
```

Comment:
First say what you want to do (explanations), then run the actual commands. Not the other way around, i.e. run commands, list explanations / rationale. Remove `$` sign, separate command from output.